core/string/string_name.h

```diff
@@ -51,6 +51,8 @@ class StringName {
 
 	struct _Data {
 		SafeRefCount refcount;
+		std::atomic<uint32_t> static_count;
```

Comment:
Why not `SafeNumeric<uint32_t>`?

It provides out of the box the same memory orderings that are used with `std::atomic` along this PR, while being a Godot-style class.

Reply:
Sorry completely forgot about it, just implemented it with this.